### PR TITLE
Utilize initializer lists

### PR DESCRIPTION
**Pull request by dschwen**
Net reduction of over 100 lines of code.

Refs #7173

### CODE REVIEW

File: framework/src/base/BoundaryRestrictable.C [hunk 1]
```diff
@@ -167,9 +167,7 @@ BoundaryRestrictable::hasBoundary(BoundaryName name) const
 {
   // Create a vector and utilize the getBoundaryIDs function, which
   // handles the ANY_BOUNDARY_ID (getBoundaryID does not)
-  std::vector<BoundaryName> names(1);
-  names[0] = name;
-  return hasBoundary(_bnd_mesh->getBoundaryIDs(names));
+  return hasBoundary(_bnd_mesh->getBoundaryIDs({name}));
```
Comment (friedmud) [inline framework/src/base/BoundaryRestrictable.C]: Awesome!

File: framework/src/bcs/NodalNormalBC.C [hunk 2]
```diff
@@ -22,9 +22,9 @@ InputParameters validParams<NodalNormalBC>()
   params.addCoupledVar("ny", "y-component of the normal");
   params.addCoupledVar("nz", "z-component of the normal");
 
-  params.set<std::vector<VariableName> >("nx") = std::vector<VariableName>(1, "nodal_normal_x");
-  params.set<std::vector<VariableName> >("ny") = std::vector<VariableName>(1, "nodal_normal_y");
-  params.set<std::vector<VariableName> >("nz") = std::vector<VariableName>(1, "nodal_normal_z");
+  params.set<std::vector<VariableName> >("nx") = {"nodal_normal_x"};
```
Comment (friedmud) [inline framework/src/bcs/NodalNormalBC.C]: So much better!

File: framework/src/multiapps/MultiApp.C [hunk 3]
```diff
@@ -235,7 +235,7 @@ MultiApp::fillPositions()
   }
   else
   {
-    _positions = std::vector<Point>(1, Point());
+    _positions = {Point()};
```
Comment (friedmud) [inline framework/src/multiapps/MultiApp.C]: Fancy!

File: framework/src/utils/RankFourTensor.C [hunk 4]
```diff
@@ -717,41 +717,24 @@ RankFourTensor::fillAntisymmetricIsotropicFromInputVector(const std::vector<Real
...
-  input9[1] = input[1];  // C1122
-  input9[2] = input[2];  // C1133
-  input9[3] = input[0];  // C2222
-  input9[4] = input[2];  // C2233 = C1133
-  input9[5] = input[3];  // C3333
-  input9[6] = input[4];  // C2323
-  input9[7] = input[4];  // C3131 = C2323
-  input9[8] = (input[0]-input[1])*0.5;  // C1212
-  fillSymmetricFromInputVector(input9, false);
+  fillSymmetricFromInputVector({input[0], input[1], input[2], input[0], input[2], input[3],
```
Comment (friedmud) [inline framework/src/utils/RankFourTensor.C]: Hmmm - looks like we lost some documentation here...

File: modules/navier_stokes/src/materials/NavierStokesMaterial.C [hunk 5]
```diff
@@ -62,7 +62,7 @@ NavierStokesMaterial::NavierStokesMaterial(const InputParameters & parameters) :
 
     // Energy equation inviscid flux matrices, "cal E_{kl}" in the notes.
     _calE(declareProperty<std::vector<std::vector<RealTensorValue> > >("calE")),
-    _vel_grads(3),
+    _vel_grads({&_grad_u, &_grad_v, &_grad_w}),
```
Comment (friedmud) [inline modules/navier_stokes/src/materials/NavierStokesMaterial.C]: Perfect!

File: modules/phase_field/src/mesh/EBSDMesh.C [hunk 6]
```diff
@@ -52,16 +52,12 @@ EBSDMesh::readEBSDHeader()
...
-  labels.push_back("x_step"); // 0
-  labels.push_back("x_dim");
-  labels.push_back("y_step"); // 2
-  labels.push_back("y_dim");
-  labels.push_back("z_step"); // 4
-  labels.push_back("z_dim");
-  labels.push_back("x_min");  // 6
-  labels.push_back("y_min");
-  labels.push_back("z_min");
+  std::vector<std::string> labels = {
```
Comment (friedmud) [inline modules/phase_field/src/mesh/EBSDMesh.C]: 👍

File: modules/xfem/src/efa/EFAElement3D.C [hunk 7]
```diff
@@ -22,15 +22,15 @@ EFAElement3D::EFAElement3D(unsigned int eid, unsigned int n_nodes, unsigned int
     EFAElement(eid, n_nodes),
     _num_faces(n_faces),
     _faces(_num_faces, NULL),
-    _face_neighbors(_num_faces, std::vector<EFAElement3D*>(1,NULL))
+    _face_neighbors(_num_faces, {NULL})
```
Comment (friedmud) [inline modules/xfem/src/efa/EFAElement3D.C]: This is kind of cool

File: framework/src/utils/RankFourTensor.C (hunk 4 above)
Comment (dschwen) [inline framework/src/utils/RankFourTensor.C]: Oh, right. I'll add that back in.

File: modules/chemical_reactions/src/actions/AddCoupledSolidKinSpeciesKernelsAction.C [hunk 8]
```diff
@@ -104,22 +104,21 @@ AddCoupledSolidKinSpeciesKernelsAction::act()
...
+        _problem->addKernel("CoupledBEKinetic", vars[i] + "_"+solid_kin_species[j] + "_kin", params_kin);
 
-        _console << vars[i]+"_"+solid_kin_species[j]+"_kin" << "\n";
+        _console << vars[i] << "_" << solid_kin_species[j] << "_kin" << "\n";
         params_kin.print();
       }
     }
-    _console << "\n";
+    _console << '\n';
   }
```
Comment (permcody) [inline modules/chemical_reactions/src/actions/AddCoupledSolidKinSpeciesKernelsAction.C]: pedantic!

**Comment by permcody** [comment]
Derek is on vacation. I'll merge this.

👍

**Pull request merged**